interpreter/python/decode.go

```diff
@@ -100,3 +106,95 @@ func decodeStubArgumentWrapperARM64(code []byte, argNumber uint8, _,
 
 	return libpf.SymbolValueInvalid
 }
+
+func decodeStubArgumentAMD64(code []byte, codeAddress, memoryBase uint64) (
+	libpf.SymbolValue, error) {
+	targetRegister := x86asm.RDI
+
+	instructionOffset := 0
+	regs := amd.RegsState{}
+
+	for instructionOffset < len(code) {
```

Comment:
The following section is focused on 64 bit code. Should we have a fallback for 32 bit code?

The original Zydis implementation was able to differentiate here, if I'm following along correctly:

```
[..]
  ZydisRegister target_register64, target_register32;
  switch (argument_no) {
  case 0:
    target_register64 = ZYDIS_REGISTER_RDI;
    target_register32 = ZYDIS_REGISTER_EDI;
    break;
[..]
```

Reply:
This works as intended with no special case handling

`regIndex` returns the same index for argno=0 (rdi/edi)
```go
	case x86asm.RDI, x86asm.EDI:
		return 5
```

I've added a synthetic test for this just in case https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/412/commits/6573aef12f480337ff9f067efd086a70a3b570c3 .  I am not aware of any actual python binaries taking advantage of this though.